data-prepper-plugins/s3-sink/src/test/java/org/opensearch/dataprepper/plugins/sink/accumulator/InMemoryBufferTest.java

```diff
@@ -0,0 +1,122 @@
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+
+class InMemoryBufferTest {
+
+    private static final String DEFAULT_CODEC_FILE_EXTENSION = "json";
+    private S3SinkConfig s3SinkConfig;
+    private ThresholdOptions thresholdOptions;
+    private BucketOptions bucketOptions;
+    private AwsAuthenticationOptions awsAuthenticationOptions;
+    private AwsCredentialsProvider awsCredentialsProvider;
+    private ObjectKeyOptions objectKeyOptions;
+    private JsonCodec codec;
+    private PluginSetting pluginSetting;
+    private PluginFactory pluginFactory;
+    private PluginModel pluginModel;
+
+    @BeforeEach
+    void setUp() throws Exception {
+
+        s3SinkConfig = mock(S3SinkConfig.class);
+        thresholdOptions = mock(ThresholdOptions.class);
+        bucketOptions = mock(BucketOptions.class);
+        awsAuthenticationOptions = mock(AwsAuthenticationOptions.class);
+        codec = mock(JsonCodec.class);
+        objectKeyOptions = mock(ObjectKeyOptions.class);
+
+        pluginSetting = mock(PluginSetting.class);
+        pluginModel = mock(PluginModel.class);
+        pluginFactory = mock(PluginFactory.class);
+
+        when(s3SinkConfig.getThresholdOptions()).thenReturn(thresholdOptions);
+        when(s3SinkConfig.getThresholdOptions().getEventCount()).thenReturn(100);
+        when(s3SinkConfig.getThresholdOptions().getMaximumSize()).thenReturn(ByteCount.parse("1kb"));
+        when(s3SinkConfig.getThresholdOptions().getEventCollect()).thenReturn(Duration.ofSeconds(5));
+
+        when(objectKeyOptions.getNamePattern()).thenReturn("my-elb-%{yyyy-MM-dd'T'hh-mm-ss}");
+
+        when(s3SinkConfig.getAwsAuthenticationOptions()).thenReturn(awsAuthenticationOptions);
+        when(awsAuthenticationOptions.getAwsRegion()).thenReturn(Region.of("us-east-1"));
+        when(awsAuthenticationOptions.authenticateAwsConfiguration()).thenReturn(awsCredentialsProvider);
+
+        when(s3SinkConfig.getBucketOptions()).thenReturn(bucketOptions);
+        when(s3SinkConfig.getBucketOptions().getObjectKeyOptions()).thenReturn(objectKeyOptions);
+        when(s3SinkConfig.getBucketOptions().getBucketName()).thenReturn("dataprepper");
+        when(s3SinkConfig.getBucketOptions().getObjectKeyOptions().getPathPrefix()).thenReturn("logdata/");
+
+        when(s3SinkConfig.getCodec()).thenReturn(pluginModel);
+        when(pluginModel.getPluginName()).thenReturn("json");
+        when(pluginFactory.loadPlugin(Codec.class, pluginSetting)).thenReturn(codec);
+
+        when(pluginSetting.getName()).thenReturn("s3");
+        when(pluginSetting.getPipelineName()).thenReturn("S3-sink-pipeline");
+    }
+
+    @Test
+    void test_in_memoryAccumulate_with_s3Upoload_success() throws InterruptedException {
+        S3SinkService s3SinkService = new S3SinkService(s3SinkConfig);
+        S3Client s3Client = s3SinkService.createS3Client();

```

Comment:
You should use a mock object here. This will let you verify the interactions with the S3Client.

Reply:
Mocked the `s3SinkService` and verified the interactions with the S3Client.